src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -215,6 +215,10 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
+        if (strlen($command->index) > 64) {
+            throw new \RunTimeException('Index names longer than 64 characters are not supported by MySQL.');
```

Comment:
Based upon other classes - would be better to use `use RunTimeException` at top of class, then just `throw new RunTimeException (...` here

Same as in the other places this occurs in the other classes?